File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n <= 2) {
            return n;
        }

        int s1 = 1;
        int s2 = 2;
        int curr = s1 + s2;
        for (int i = 3; i <= n; i++) {
            curr = s1 + s2;
            s1 = s2;
            s2 = curr;
        }

        return curr;
    }
}

File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        if (n <= 3) {
            return n;
        }

        int a = 1, b = 2;
        int c = 3;
        for (int i = 3; i <= n; i++) {
            c = a + b;
            a = b;
            b = c;
        }
        return c;
    }
};